frontend/hub/collections/CollectionDetails.tsx

```diff
@@ -55,20 +56,92 @@ import { HubItemsResponse } from '../useHubView';
 import { useCollectionActions } from './hooks/useCollectionActions';
 import { useCollectionColumns } from './hooks/useCollectionColumns';
+import { PageSingleSelect } from './../../../framework/PageInputs/PageSingleSelect';
+import { CheckCircleIcon, ExclamationTriangleIcon } from '@patternfly/react-icons';
+import { Label } from '@patternfly/react-core';
+import { AwxError } from '../../awx/common/AwxError';
 
 export function CollectionDetails() {
   const { t } = useTranslation();
-  const params = useParams<{ name: string; namespace: string; repository: string }>();
-  const { data, refresh } = useGet<HubItemsResponse<CollectionVersionSearch>>(
-    hubAPI`/v3/plugin/ansible/search/collection-versions/?name=${params.name || ''}&namespace=${
-      params.namespace || ''
-    }&repository=${params.repository || ''} }`
+  const [searchParams, setSearchParams] = useSearchParams();
+  const itemActions = useCollectionActions(() => void refresh());
+  const [collection, setCollection] = useState<CollectionVersionSearch | undefined>(undefined);
+
+  let collectionError: JSX.Element | undefined = undefined;
+
+  function setVersionParams(version: string) {
+    setSearchParams((params) => {
+      params.set('version', version);
+      return params;
+    });
+  }
+  // load all collections versions belong to the repository
+  const collectionsResult = useGet<HubItemsResponse<CollectionVersionSearch>>(
+    hubAPI`/v3/plugin/ansible/search/collection-versions/?name=${
+      searchParams.get('name') || ''
+    }&namespace=${searchParams.get('namespace') || ''}&repository_name=${
+      searchParams.get('repository') || ''
+    }&order_by=-pulp_created`
   );
-  let collection: CollectionVersionSearch | undefined = undefined;
-  if (data && data.data && data.data.length > 0) {
-    collection = data.data[0];
+
+  if (
+    collectionsResult.error ||
+    (collectionsResult.isLoading == false &&
+      collectionsResult.data &&
+      collectionsResult.data.data.length == 0)
+  ) {
+    collectionError = (
+      <AwxError error={collectionsResult.error || { name: 'not found', message: t('Not Found') }} />
+    );
   }
-  const itemActions = useCollectionActions(() => void refresh());
+
+  const collections = collectionsResult.data ? collectionsResult.data.data : [];
+
+  // load collection by search params
+  const version = searchParams.get('version');
+
+  let highestFilter = '';
+  let versionFilter = '';
+
+  if (!version) {
+    // for unspecified version, load highest
+    highestFilter = '&isHighest=true';
+  } else {
+    versionFilter = '&version=' + version;
+  }
+
+  const { data, refresh, error, isLoading } = useGet<HubItemsResponse<CollectionVersionSearch>>(
+    hubAPI`/v3/plugin/ansible/search/collection-versions/?name=${
+      searchParams.get('name') || ''
+    }&namespace=${searchParams.get('namespace') || ''}&repository_name=${
+      searchParams.get('repository') || ''
+    }&order_by=-pulp_created` +
```

Comment:
(and here)

Reply:
I hope this will always return 1 item anyway :)